gcp/tx-relay/src/Signer.ts

```diff
@@ -0,0 +1,177 @@
+    [Address.settleAccounts_ARBITRUM]: [Sign.settleAccounts, Sign.settleVaultsAccounts],
+  }
+}
+
+const keysToUse = {
+  [Address.TREASURY_MANAGER]: Key.reinvestment,
+  [Address.AaveV3Pool_ARBITRUM]: Key.liquidation,
+  [Address.AaveV3Pool_MAINNET]: Key.liquidation,
+  [Address.AaveFlashLiquidator_ARBITRUM]: Key.liquidation,
+  [Address.AaveFlashLiquidator_MAINNET]: Key.liquidation,
+  [Address.Multicall3]: Key.liquidation,
+  [Address.initializeAllMarkets_ARBITRUM]: Key.liquidation,
+  [Address.settleAccounts_ARBITRUM]: Key.liquidation,
+};
+
+function getTxType({ signature }: { signature: Sign }) {
+  if ([Sign.flashLoan, Sign.flashLiquidate, Sign.settleVaultsAccounts, Sign.settleAccounts, Sign.initializeAllMarkets].includes(signature)) {
+    return 'liquidation';
+  }
+  if ([Sign.claimVaultRewardTokens, Sign.reinvestVaultReward].includes(signature)) {
+    return 'reinvestment';
+  }
+  if ([Sign.executeTrade, Sign.investWETHAndNOTE].includes(signature)) {
+    return 'burnNote';
+  }
+  if ([Sign.harvestAssetsFromNotional].includes(signature)) {
+    return 'harvesting';
+  }
+  if ([Sign.settleVaultsAccounts, Sign.settleAccounts, Sign.initializeAllMarkets].includes(signature)) {
+    return 'initialize-markets';
+  }
+  return 'unknown';
+}
+
+async function logToDataDog(message: any) {
+  return fetch("https://http-intake.logs.datadoghq.com/api/v2/logs", {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+      'DD-API-KEY': process.env.DD_API_KEY as string
+    },
+    body: JSON.stringify({
+      ddsource: "tx-relay",
+      service: "signer",
+      message
+    })
+  }).catch(err => console.error(err));
+}
+
+// will only work properly if node app is not run in cluster
+const throttle = (() => {
+  const wait = (ms: number) => new Promise(r => setTimeout(r, ms));
+  const delay = 2000;
+  const lastCallPerDomain = new Map();
+
+  return async (domain: string) => {
+    while (Date.now() < (lastCallPerDomain.get(domain) || 0) + delay) {
+      await wait(delay - (Date.now() - (lastCallPerDomain.get(domain) || 0)));
+    }
+    lastCallPerDomain.set(domain, Date.now());
+  };
+})();
+
+export async function sendTransaction({ to, data, gasLimit, network }: { to: string, data: string, gasLimit: number, network: string }) {
+  to = to.toLowerCase();
+  const signature = data.slice(0, 10) as Sign;
+  if (
+    !whitelist[network][to] ||
+    !whitelist[network][to].includes(signature)
+  ) {
+    throw new Error(`Call to: ${to} method: ${signature} on network: ${network} is not allowed`);
+  }
+
+  // in order to properly fetch correct nonce for tx we need to throttle here per key
+  await throttle(keysToUse[to]);
+
+  const provider = ethers.getDefaultProvider(rpcUrls[network]);
+
+  const signer = new GcpKmsSigner({ ...kmsCredentials, keyId: keysToUse[to] }).connect(provider);
+
+  const tx = await signer.sendTransaction({
```

Comment:
@sbuljac if this throws an error can we log a special transaction failed event into datadog? I will set up a monitor to alert on that